ev3dev2/sensor/lego.py

```diff
@@ -660,6 +674,61 @@ def wait_until_angle_changed_by(self, delta, direction_sensitive=False):
             while abs(start_angle - self.value(0)) < delta:
                 time.sleep(0.01)
 
+    def circle_angle(self):
+        """
+        As the gryo rotates clockwise the angle increases, it will increase
+        by 360 for each full rotation. As the gyro rotates counter-clockwise
+        the gyro angle will decrease.
+
+        The angles on a circle have the opposite behavior though, they start
+        at 0 and increase as you move counter-clockwise around the circle.
+
+        Convert the gyro angle to the angle on a circle. We consider the initial
+        position of the gyro to be at 90 degrees on the cirlce.
+        """
+        current_angle = self.angle
+        delta = abs(current_angle - self._init_angle) % 360
+
+        if delta == 0:
```

Comment:
This seems over-complicated. Shouldn't the transformation from gyro angle to appropriate angle always be `some_constant_offset - current_angle`?

Reply:
I thought so too, figured it would be relatively simple.  I spent a few hours coming up with this function though...maybe I am missing something.